bluesky/magics.py

```diff
@@ -110,53 +113,161 @@ def ct(self, line):
-                value = np.round(v, decimals=prec)
-                try:
-                    low_limit, high_limit = p.limits
-                except Exception as exc:
-                    low_limit = high_limit = exc.__class__.__name__
-                else:
-                    low_limit = np.round(low_limit, decimals=prec)
-                    high_limit = np.round(high_limit, decimals=prec)
-                try:
-                    offset = p.user_offset.get()
-                except Exception as exc:
-                    offset = exc.__class__.__name__
-                else:
-                    offset = np.round(offset, decimals=prec)
+                low_limit = high_limit = exc.__class__.__name__
             else:
-                value = v.__class__.__name__  # e.g. 'DisconnectedError'
-                low_limit = high_limit = offset = ''
+                low_limit = np.round(low_limit, decimals=prec)
+                high_limit = np.round(high_limit, decimals=prec)
+            try:
+                offset = p.user_offset.get()
+            except Exception as exc:
+                offset = exc.__class__.__name__
+            else:
+                offset = np.round(offset, decimals=prec)
+        else:
+            value = v.__class__.__name__  # e.g. 'DisconnectedError'
+            low_limit = high_limit = offset = ''
+
+        lines.append(LINE_FMT.format(p.name, value, low_limit, high_limit,
+                                     offset))
+    print('\n'.join(lines))
+
+
+def _labeled_devices(user_ns=None, maxdepth=6):
+    ''' Returns list of all devices that are labeled.
+
+        Parameters
+        ----------
+        user_ns : dict, optional
+            The namespace to search on
+
+        maxdepth: int, optional
+            max recursion depth
+
+        Returns
+        -------
+            A dictionary of (name, ophydobject) tuple indexed by device label.
+
+        Examples
+        --------
+        Read devices labeled as motors:
+            objs = _labeled_devices()
+            my_motors = objs['motors']
+    '''
+    # could be set but lists are more common for users
+    obj_list = collections.defaultdict(list)
+
+    if maxdepth <= 0:
+        print("Recursion limit exceeded")
+        return obj_list
+
+    if user_ns is None:
+        user_ns = get_ipython().user_ns
+
+    for key, obj in user_ns.items():
+        # ignore objects beginning with "_"
+        # (mainly for ipython stored objs from command line
+        # return of commands)
+        # also check its a subclass of desired classes
+        if not key.startswith("_"):
+            if is_parent(obj):
+                labels = getattr(obj, '_ophyd_labels_', set())
+                obj_list.update(_labeled_devices(user_ns=obj.__dict__,
+                                              maxdepth=maxdepth-1,))
+            else:
+                if hasattr(obj, '_ophyd_labels_'):
+                    # inherit parent labels
```

Comment:
The current code does _not_ inherit parent labels, right? I _think_ that's probably the best behavior. (I suppose we could have a keyword arg to turn propagation on or off, but I don't think we necessarily need to build that until we know for sure that we need it for something.) Anyway, is this comment vestigial?

Reply:
oops! fixed